### PR TITLE
Updates to OpenCG compatibility module to reflect move to PEP8 compliance

### PR DESCRIPTION
**Pull request by wbinventor**
This PR updates the OpenCG compatibility module to reflect @!paulromano's recent work to make OpenCG PEP8 compliant. In particular, OpenCG's class methods and other routines now use underscores "_" rather than camelCase notation, and many getters/setters have been replaced with Python property decorators.

### CODE REVIEW

File: openmc/opencg_compatible.py [hunk 1]
```diff
@@ -196,59 +196,59 @@ def get_opencg_surface(openmc_surface):
...
-    if openmc_surface._type == 'plane':
-        A = openmc_surface._coeffs['A']
-        B = openmc_surface._coeffs['B']
-        C = openmc_surface._coeffs['C']
-        D = openmc_surface._coeffs['D']
+    if openmc_surface.type == 'plane':
+        A = openmc_surface.coeffs['A']
+        B = openmc_surface.coeffs['B']
+        C = openmc_surface.coeffs['C']
+        D = openmc_surface.coeffs['D']
```
Comment (paulromano) [inline openmc/opencg_compatible.py]: Any reason you don't use the `a`, `b`, `c`, and `d` properties directly? Same comment goes for all the other surfaces.

File: openmc/opencg_compatible.py [hunk 2]
```diff
@@ -455,37 +455,37 @@ def get_opencg_cell(openmc_cell):
...
-    fill = openmc_cell._fill
+    fill = openmc_cell.fill
 
-    if (openmc_cell._type == 'normal'):
-        opencg_cell.setFill(get_opencg_material(fill))
-    elif (openmc_cell._type == 'fill'):
-        opencg_cell.setFill(get_opencg_universe(fill))
+    if (openmc_cell.fill_type == 'material'):
+        opencg_cell.fill = get_opencg_material(fill)
+    elif (openmc_cell.fill_type == 'universe'):
```
Comment (paulromano) [inline openmc/opencg_compatible.py]: Shouldn't this be `openmc_cell.type`? And actually, looking at the source, it looks like there's a bug in that property (it returns `self._fill`. Can you fix that?

**Comment by paulromano** [comment]
Looks good other than those two comments.

File: openmc/opencg_compatible.py (hunk 2 above)
Comment (wbinventor) [inline openmc/opencg_compatible.py]: Yes, I actually had fixed this on my MGXS branch and was going to submit the fix there, but I patched it in here instead. I've replaced the `Cell.type` property with a `Cell.fill_type` property which returns a string "material," "universe" or "lattice."

**Comment by wbinventor** [comment]
This has been updated to address your comments @paulromano

**Pull request merged**